crates/bevy_reflect/derive/src/from_reflect.rs

```diff
@@ -27,14 +27,21 @@ pub(crate) fn impl_opaque(meta: &ReflectMeta) -> proc_macro2::TokenStream {
     let bevy_reflect_path = meta.bevy_reflect_path();
     let (impl_generics, ty_generics, where_clause) = type_path.generics().split_for_impl();
     let where_from_reflect_clause = WhereClauseOptions::new(meta).extend_where_clause(where_clause);
+
+    let exact_match = quote! {
+        if let #FQOption::Some(value) = <dyn #bevy_reflect_path::PartialReflect>::try_downcast_ref::<#type_path #ty_generics>(reflect) {
```

Comment:
I recently had to fix this function to properly account for remote types in #19158. Might be worth adding that here or waiting for that PR to be merged.

Reply:
It looks like #19158 is just about ready to be merged. I can rebase again once that's done if needed.